R/preprocessing.R

```diff
@@ -74,31 +77,46 @@ fill_missing <- function(data,
   assert_character(missing_obs)
   assert_character(obs_column)
   assert_character(by, null.ok = TRUE)
-  if (!missing(initial_accumulate)) {
-    assert_integerish(initial_accumulate, lower = 1)
-  }
   assert_names(
     colnames(data),
     must.include = c("date", by, obs_column),
     disjunct.from = "accumulate"
   )
   assert_date(data$date, any.missing = FALSE)
 
-  data <- as.data.table(data)
-
   missing_dates <- arg_match(missing_dates)
   missing_obs <- arg_match(missing_obs)
 
+  data <- as.data.table(data)
+
+  if (missing(initial_accumulate)) {
+    ## detect frequency of accumulation if possible
+    diffs <- data[, list(diff = unique(diff(date))), by = by]
```

Comment:
Struggling to parse the different uses of `diff` here so suggesting
```suggestion
    missing_date_patterns <- data[, list(pattern = unique(diff(date))), by = by]
```